### PR TITLE
improvement: adding the disabled text color, fixing inverted text colors

### PR DESCRIPTION
**Pull request by lloydaf**
**What:**
This PR adds the `disabled` state to the `Text` component, in alignment with the design for wave. It also fixes colors when `inverted` (for `weak` and `disabled` texts) also in alignment with the design.
​
**Why:**
This is in regard to [SUP-11112](https://jira.intapps.it/browse/SUP-11112)
​
**How:**
* Add a new disabled property to the `Text` component
* Fix the colors for inverted state
* Add examples to the documentation
​
**Media:**
Design
Documentation Screenshot

### CODE REVIEW

**Comment by duranmla** [comment]
I don't have any concerns with the implementation, nonetheless, I feel a bit lost with the need and if this should be something that designers actually do. Feels odd to talk about "disabled text". Besides, I will encourage you to bring these topics to further discussion with designers and the dev community at Github.

From the link, you refer to I just see you as a reporter outlining a need, but is indeed this valid? _(sorry for the doubts I just want to make sure designers and dev are align)_ @rafael-sepeda @agonzalezUX can you folks help here to answer the following question:

**Do designer have the concept of "disabled" text?**

**Comment by lloydaf** [comment]
Screenshot
Hi @rafael-sepeda, with the color codes you mentioned, this is how it looks. Does this look correct? If so I can update the PR with these new ones

**Comment by viniciusleitao** [comment]
> @duranmla , @lloydaf
> for the new setup in Figma we created this tokens:
> 
> ```
> "text": {
>     "primary": "$colors.brand.authenticBlue.900",
>     "primary-inverted": "$colors.brand.white",
>     "secondary": "$colors.brand.authenticBlue.550",
>     "secondary-inverted": "$colors.brand.authenticBlue.50",
>     "disabled": "$colors.brand.authenticBlue.350",
>     "disabled-inverted": "$colors.brand.authenticBlue.200",
>     "link": "$colors.brand.actionBlue.900",
>     "link-inverted": "$colors.brand.white",
>     "danger": "$colors.brand.negativeOrange.900",
>     "success": "$colors.brand.positiveGreen.900",
>     "warning": "$colors.brand.attentionYellow.900"
> },
> ```
> 
> We where using before the `weak` but now renamed it to `secondary`

@rafael-sepeda Are we using the authenticBlue.50 for secondary-inverted now? It used to be the 200, right? I'm not really sure if the contrast between primary and secondary in the inverted colors will match the default colors in this new setup. Is there a reason why we changed it?

**Comment by rafael-sepeda** [comment]
@agonzalezUX Could you check the colors?

**Comment by agonzalezUX** [comment]
Hi guys, the color codes @rafael-sepeda shared with you are correct

Regarding Vini's comment, he is right, the secondary-inverted should not be authenticBlue.50, there is a mistake and it should be authenticBlue.550

Regarding the suggestion @viniciusleitao is raising regarding the usage of authenticBlue.200 for the secondary inverted, I have to say that in wave we never differentiated between secondary default and inverted as both font color used the same value. However I am open to have this discussion between product designers.

So, in conclusion we should use:
```
"text": {
    "primary": "$colors.brand.authenticBlue.900",
    "primary-inverted": "$colors.brand.white",
    "secondary": "$colors.brand.authenticBlue.550",
    "secondary-inverted": "$colors.brand.authenticBlue.550",
    "disabled": "$colors.brand.authenticBlue.350",
    "disabled-inverted": "$colors.brand.authenticBlue.200",
    "link": "$colors.brand.actionBlue.900",
    "link-inverted": "$colors.brand.white",
    "danger": "$colors.brand.negativeOrange.900",
    "success": "$colors.brand.positiveGreen.900",
    "warning": "$colors.brand.attentionYellow.900"
},
```

**Comment by lloydaf** [comment]
Screenshot
This is how it looks, @agonzalezUX, updating the PR with these values

**Comment by viniciusleitao** [comment]
For me "secondary-inverted" and "disabled-inverted" colors are inverted. With the current implementation, Disabled looks brighter than Weak and it should be the other way around.

**Comment by agonzalezUX** [comment]
Sorry guys I think @viniciusleitao is right. Default should go from darker to lighter and inverted form lighter to darker. Sorry for the confusion 🙏🏼

please @viniciusleitao and @rafael-sepeda double check those are the correct values:
```
"text": {
    "primary": "$colors.brand.authenticBlue.900",
    "primary-inverted": "$colors.brand.white",
    "secondary": "$colors.brand.authenticBlue.550",
    "secondary-inverted": "$colors.brand.authenticBlue.350",
    "disabled": "$colors.brand.authenticBlue.350",
    "disabled-inverted": "$colors.brand.authenticBlue.550",
    "link": "$colors.brand.actionBlue.900",
    "link-inverted": "$colors.brand.white",
    "danger": "$colors.brand.negativeOrange.900",
    "success": "$colors.brand.positiveGreen.900",
    "warning": "$colors.brand.attentionYellow.900"
},
```

**Comment by lloydaf** [comment]
As we have ended up with the initial values in the PR, I am removing all the commits in between to make our lives easier :)

**Comment by nlopin** [comment]
@rafael-sepeda @viniciusleitao @agonzalezUX I recommend adding some guidance on when to use this type of text. It definitely fails to pass any a11y contrast checks. This is one of the key elements to make the page perceivable by everybody.

I think it is important to put to the document the following info:

* [WCAG-2.1 recommendation](https://www.w3.org/WAI/WCAG21/Understanding/contrast-enhanced) is to keep enough color contrast for all the elements except components in the inactive/disabled state and pure decoration texts.
* [Official contrast checker](https://webaim.org/resources/contrastchecker/) and recommendation to use at least AA level of contrast
* Figma users can install [Constrast plugin](https://www.figma.com/community/plugin/748533339900865323/Contrast) or [A11y - Color Contrast Checker](https://www.figma.com/community/plugin/733159460536249875/A11y---Color-Contrast-Checker)

**Comment by viniciusleitao** [comment]
> It definitely fails to pass any a11y contrast checks

@nlopin just to clarify, the only color combination that does not pass the contrast check is the disabled one. Checking other design systems I realized this is some kind of pattern on disabled states.

- [Material](https://material.io/components/text-fields#outlined-text-field)
- [Atalassian](https://atlassian.design/components/checkbox/examples)
- [Carbon](https://www.carbondesignsystem.com/components/text-input/usage/)

**Comment by nlopin** [comment]
@viniciusleitao yes, there are no contrast requirements for the disabled states

**Comment by rafael-sepeda** [comment]
@lloydaf , @nlopin , @agonzalezUX , @viniciusleitao 
Thanks for pushing this. I hope that soon we will be able to automate the color sync between Figma and wave. We discussed already that we can start with semantic colors and see how we can consume the color tokens definition in JSON  in code and extract it so we can import it back to Figma.

**Review by duranmla — APPROVED** [review]

**Comment by lloydaf** [comment]
thanks @duranmla ❤️
What are the next steps ideally? Should I merge after I get enough approvals or do you merge on behalf of me? Both are fine, as long as we merge this before #97. Right now it says 
> Only those with write access to this repository can merge pull requests.

**Comment by nlopin** [comment]
One of the code owners will merge to main and the new version will be published automatically. I think we want one approval from the design team

**Comment by lloydaf** [comment]
@nlopin what documentation page would you want to modify/add and what specific information would you like represented in the docs? I can try and add that if you want.

**Review by snapsnapturtle — APPROVED** [review]

**Pull request merged**
